// @akashic-extension/akashic-label/lib/FragmentDrawInfo.js
window.gLocalAssetContainer["node_modules/@akashic-extension/akashic-label/lib/FragmentDrawInfo.js"] = function(g) { (function(exports, require, module, __filename, __dirname) {
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RubyFragmentDrawInfo = exports.StringDrawInfo = void 0;
/**
 * 行に含まれる文字列要素。
 */
class StringDrawInfo {
    constructor(text, width, glyphs) {
        this.text = text;
        this.width = width;
        this.glyphs = glyphs;
    }
}
exports.StringDrawInfo = StringDrawInfo;
/**
 * 行に含まれるルビ要素。
 */
class RubyFragmentDrawInfo {
    constructor(fragment, width, rbWidth, rtWidth, glyphs, rubyGlyphs) {
        this.text = fragment.text;
        this.fragment = fragment;
        this.width = width;
        this.rbWidth = rbWidth;
        this.rtWidth = rtWidth;
        this.glyphs = glyphs;
        this.rubyGlyphs = rubyGlyphs;
    }
}
exports.RubyFragmentDrawInfo = RubyFragmentDrawInfo;

})(g.module.exports, g.module.require, g.module, g.filename, g.dirname);
}
